src/sentry/sentry_apps/apps.py

```diff
@@ -295,6 +297,11 @@ def run(self, *, user: User | RpcUser, request: HttpRequest | None = None) -> Se
     def _generate_and_validate_slug(self) -> str:
         slug = generate_slug(self.name, is_internal=self.is_internal)
 
+        # If option is set, add random 3 lowercase letter suffix to prevent numeric slug
+        # eg: 123 -> 123-abc
+        if options.get("api.prevent-numeric-slugs") and slug.isnumeric():
+            slug = f"{slug}-{''.join(random.choice(string.ascii_lowercase) for _ in range(3))}"
```

Comment:
nonblocking question: are we always doing this to prevent numeric slugs? maybe pull into a util function?

Reply:
So far I'm gonna be doing this in two places, this one here and another PR today for the DocIntegrationsEndpoint POST. I think it's okay to not use a util function if it's only in 2 places